src/view/editor/parts/recurrence/views/weekly-options.tsx

```diff
@@ -0,0 +1,289 @@
+
+const RADIO_VALUES = {
+	QUICK_OPTIONS: 'QuickOptions',
+	CUSTOM_OPTIONS: 'CustomOptions'
+};
+
+const defaultState = {
+	freq: 'WEE',
+	interval: {
+		ival: 1
+	},
+	byday: {
+		wkday: [{ day: 'MO' }]
+	},
+	radioValue: RADIO_VALUES.QUICK_OPTIONS
+};
+
+const radioInitialState = (
+	freq: string | undefined,
+	interval: Interval | undefined,
+	byday: Byday | undefined,
+	workingSchedule: WorkWeekDay[]
+): string => {
+	if ((freq === 'DAI' || freq === 'WEE') && byday?.wkday) {
+		// building an array with the same structure of wkday to check if they have the same values
+		// to determine if we are receiving the value of the working day option
+		const workingDays = map(filter(workingSchedule, ['working', true]), (workingDay) => {
+			const day = find(WEEK_SCHEDULE, ['value', workingDay.day])?.label?.slice(0, 2);
+			return { day };
+		});
+		const diff = differenceWith(workingDays, byday?.wkday, isEqual);
+		if (diff?.length === 0 && interval?.ival === 1) {
+			return RADIO_VALUES.QUICK_OPTIONS;
+		}
+		return RADIO_VALUES.CUSTOM_OPTIONS;
+	}
+	if (freq === 'WEE') {
+		if (interval?.ival === 1) {
+			return RADIO_VALUES.QUICK_OPTIONS;
+		}
+		return RADIO_VALUES.CUSTOM_OPTIONS;
+	}
+	return defaultState.radioValue;
+};
+
+const startValueInitialState = (
+	freq: string | undefined,
+	interval: Interval | undefined,
+	byday: Byday | undefined,
+	weeklyOptions: { label: string; value: string }[]
+): RecurrenceStartValue | undefined => {
+	if (freq === 'WEE') {
+		return omitBy({ interval, byday }, isNil);
+	}
+	if (freq === 'DAI' && byday?.wkday && interval?.ival === 1) {
+		const option = find(
+			weeklyOptions,
+			(item) =>
+				differenceWith(
+					map(item?.value?.split?.(','), (day) => ({ day })),
+					byday?.wkday ?? {},
+					isEqual
+				).length === 0
+		);
+		if (option) {
+			return { interval, byday };
+		}
+		return { interval: defaultState.interval, byday: defaultState.byday };
+	}
+	return undefined;
+};
+
+const selectInitialValue = (
+	weeklyOptions: { label: string; value: string }[],
+	byday: Byday | undefined
+): { label: string; value: string } =>
+	find(
+		weeklyOptions,
+		(item) =>
+			differenceWith(
+				map(item?.value?.split?.(','), (day) => ({ day })),
+				byday?.wkday ?? [],
+				isEqual
+			).length === 0
+	) ?? weeklyOptions[0];
+
+const WeeklyOptions = ({ editorId }: { editorId: string }): ReactElement | null => {
+	const { frequency, setNewStartValue } = useContext(RecurrenceContext);
+	const freq = useSelector(selectEditorRecurrenceFrequency(editorId));
+	const interval = useSelector(selectEditorRecurrenceInterval(editorId));
+	const byday = useSelector(selectEditorRecurrenceByDay(editorId));
+
+	const prefs = usePrefs();
+
+	const workingSchedule = useMemo(
+		() => workWeek(prefs.zimbraPrefCalendarWorkingHours),
+		[prefs?.zimbraPrefCalendarWorkingHours]
+	);
+
+	const weekOptions = useMemo(
+		() => [
+			{ label: t('label.week_day.monday', 'Monday'), value: 'MO' },
```

Comment:
The same definitions are duplicate in other files. Consider the possibility to move them in a common file and importing them where needed

Reply:
Yeah, It was something I had to do and at the end I forgot about it. I pushed all the changes, please check last commit